peer.go

```diff
@@ -564,17 +564,28 @@ func (p *peer) loadActiveChannels(chans []*channeldb.OpenChannel) (
 		peerLog.Tracef("Using link policy of: %v",
 			spew.Sdump(forwardingPolicy))
 
-		// Register this new channel link with the HTLC Switch. This is
-		// necessary to properly route multi-hop payments, and forward
-		// new payments triggered by RPC clients.
+		// If the channel is pending, set the value to nil in the
+		// activeChannels map. This is done to signify that the channel is
+		// pending. We don't add the link to the switch here - it's the funding
+		// manager's responsibility to spin up pending channels. Adding them
+		// here would just be extra work as we'll tear them down when creating
+		// + adding the final link.
+		if lnChan.IsPending() {
+			p.activeChanMtx.Lock()
+			p.activeChannels[chanID] = nil
```

Comment:
Have you considered using a distinct map here instead of setting the value to `nil`? Typically the default value will be returned if the key isn't set at all. However in this case, since we start to set the `nil` value _ourselves_, all callers will need to always check: `nil && !ok` instead of just `!ok`. All iteration over this map will now also need to handle this case as well.

Reply:
Yes I have considered using a distinct map and I agree with your point that it makes iteration more complicated. Does having a new map make it more or less complicated?